packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/relational/RelationalInstanceSetImplementationState.ts

```diff
@@ -132,7 +141,92 @@ export class RelationalPropertyMappingState extends PropertyMappingState {
   });
 }
 
-export class RootRelationalInstanceSetImplementationState extends InstanceSetImplementationState {
+export abstract class RelationalInstanceSetImplementationState extends InstanceSetImplementationState {}
+
+export class EmbeddedRelationalInstanceSetImplementationState
+  extends RelationalInstanceSetImplementationState
+  implements RelationalPropertyMappingState
+{
+  declare instanceSetImplementationState: RelationalInstanceSetImplementationState;
+  declare mappingElement: EmbeddedRelationalInstanceSetImplementation;
+  declare propertyMapping: EmbeddedRelationalInstanceSetImplementation;
+
+  constructor(
+    editorStore: EditorStore,
+    instanceSetImplementationState: RelationalInstanceSetImplementationState,
+    setImplementation: EmbeddedRelationalInstanceSetImplementation,
+  ) {
+    super(editorStore, setImplementation);
+    this.instanceSetImplementationState = instanceSetImplementationState;
+    this.mappingElement = setImplementation;
+    this.propertyMapping = setImplementation;
+    this.propertyMappingStates = this.getPropertyMappingStates(
+      setImplementation.propertyMappings,
+    );
+  }
+
+  get lambdaId(): string {
+    throw new IllegalStateError(
+      `Can't build lambda ID for embedded relational instance set implementation state`,
+    );
+  }
+
+  getPropertyMappingStates(
+    propertyMappings: PropertyMapping[],
+  ): RelationalPropertyMappingState[] {
+    // TODO
+    return [];
+  }
+
+  // dummy lambda editor states needed because embedded flat-data should be seen as `PropertMappingState`
+  lambdaPrefix = '';
+  lambdaString = '';
+  parserError?: ParserError;
+  compilationError?: CompilationError;
+
+  decorate(): void {
+    throw new UnsupportedOperationError();
+  }
+  convertPropertyMappingTransformObjects(): Promise<void> {
+    throw new UnsupportedOperationError();
+  }
+  setLambdaString(val: string): void {
+    throw new UnsupportedOperationError();
+  }
+  setParserError(error: ParserError | undefined): void {
+    throw new UnsupportedOperationError();
+  }
+  setCompilationError(error: CompilationError | undefined): void {
+    throw new UnsupportedOperationError();
+  }
+  get fullLambdaString(): string {
+    throw new UnsupportedOperationError();
+  }
+  processSourceInformation(
```

Comment:
Fix bug leaving form mode and breaking due to one of these methods probably

Reply:
Done!